File: go-sdk/pkg/models/operations/listdatabaseregions.go
package operations

type ListDatabaseRegionsPathParams struct {
	Name         string `pathParam:"style=simple,explode=false,name=name"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListDatabaseRegionsQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListDatabaseRegionsRequest struct {
	PathParams  ListDatabaseRegionsPathParams
	QueryParams ListDatabaseRegionsQueryParams
}

type ListDatabaseRegionsResponse struct {
	ContentType                                 string
	StatusCode                                  int64
	ListDatabaseRegions200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/listdatabaseregions.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class ListDatabaseRegionsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class ListDatabaseRegionsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}

export class ListDatabaseRegionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListDatabaseRegionsPathParams;

  @SpeakeasyMetadata()
  queryParams: ListDatabaseRegionsQueryParams;
}

export class ListDatabaseRegionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listDatabaseRegions200ApplicationJSONObject?: Record<string, Record<string, any>>;
}